protocol/communities/persistence.go

```diff
@@ -1669,3 +1669,40 @@ func (p *Persistence) AllNonApprovedCommunitiesRequestsToJoin() ([]*RequestToJoi
 	}
 	return nonApprovedRequestsToJoin, nil
 }
+
+func (p *Persistence) RemoveAllCommunityRequestsToJoinExceptOur(pk string, communityID []byte) error {
```

Comment:
suggestion: it can be any pk, so `RemoveAllCommunityRequestsToJoinExceptPublicKey`